exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/UnsafeAccess.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal.marshal;
+
+import java.lang.reflect.Field;
+import sun.misc.Unsafe;
+
+class UnsafeAccess {
+  private static final boolean available = checkUnsafe();
+
+  static boolean isAvailable() {
+    return available;
+  }
+
+  private static boolean checkUnsafe() {
+    try {
```

Comment:
Let's add an escape hatch to disable unsafe access even if its available. 
```suggestion
    boolean unsafeDisabled =
        Boolean.parseBoolean(ConfigUtil.getString("otel.java.experimental.exporter.disable_unsafe", "false"));
    if (unsafeDisabled) {
      return false;
    }
    try {
```

Reply:
I used `otel.java.experimental.exporter.unsafe.enabled` as the property. Also added jvm version check so that unsafe usage is disabled by default on 23 and newer.